examples/chat-online-status/react/src/app/page.tsx

```diff
@@ -0,0 +1,95 @@
+"use client";
+
+import React, { useEffect, useState } from 'react';
+import { useChatClient, useRoom, usePresenceListener, usePresence } from '@ably/chat/react';
+import { PresenceEvent } from '@ably/chat';
+import '../../styles/styles.css';
+
+interface OnlineStatus {
+  status: string;
+}
+
+export default function Home() {
+  const { roomStatus, connectionStatus } = useRoom();
+
+  if (roomStatus !== 'attached' || connectionStatus !== 'connected') {
+    return <div>Loading...</div>;
+  }
+
+  return (
+    <div>
+      <Online />
+    </div>
+  );
+}
+
+const Online = () => {
+  const [onlineStatuses, setOnlineStatuses] = useState<PresenceEvent[]>([]);
+  const { presenceData, error } = usePresenceListener({
+    listener: (presence) => {
+      if (presence.action === 'enter') {
+        setOnlineStatuses((prevStatus: PresenceEvent[]) => [...prevStatus, presence as PresenceEvent]);
+      } else if (presence.action === 'leave') {
+        setOnlineStatuses((prevStatus: PresenceEvent[]) => prevStatus.filter((status: PresenceEvent) => status.clientId !== presence.clientId));
+      }
+    },
+  });
+
+  const { clientId } = useChatClient();
+
+  useEffect(() => {
+    if (presenceData) {
```

Comment:
Is the idea to show both methods to retrieve presence data? I'm not sure what the docs look like, but this could be confusing? Perhaps just use one and console log the other? Also maybe some comments to let the user know that both methods are valid and shown for demonstration?

Reply:
I'm sorry @splindsay-92 I'm not following. I'm using presenceData from the usePresenceListener in order to prepopulate those that are already in the presence set. Otherwise it'll just show you online and then anyone that joins the set later.